scipy/signal/signaltools.py

```diff
@@ -501,17 +621,39 @@ def convolve(in1, in2, mode='full'):
         # Convolution is commutative; order doesn't have any effect on output
         volume, kernel = kernel, volume
 
+    if method == 'fft' and not _numeric_arrays([volume, kernel]):
+        raise ValueError('convolve can only be called with method=="fft" when'
+                         ' arrays consist of numeric elements (i.e., '
+                         'int/float/etc)')
+
+    if method == 'auto':
+        method = 'fft' if _fftconv_faster(volume, kernel, mode) else 'direct'
+        method = 'direct' if not _numeric_arrays([volume, kernel]) else method
+
+    if method in ('fft', 'auto') and hasattr(np, "complex256"):
+        method = 'direct' if volume.dtype == 'complex256' else method
+        method = 'direct' if kernel.dtype == 'complex256' else method
+
+    # catch when more precision required than float provides (representing a
+    # integer as float can lose precision in fftconvolve if larger than 2**52)
+    if method == 'fft' and any([_numeric_arrays([x], kinds='ui')
+                                        for x in [volume, kernel]]):
+        max_value = int(np.abs(volume).max()) * int(np.abs(kernel).max())
+        max_value *= int(min(volume.size, kernel.size))
+        method = 'direct' if max_value > 2**np.finfo('float').nmant - 1 \
```

Comment:
if user says `method='fft'` it should either use fft or raise an exception if it can't.  it shouldn't automatically use slow direct mode if they specifically requested fft.  if they request fft and float can't reproduce ints exactly, it should still do it with inaccurate float anyway, the way fftconvolve does.